jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecr/actions/PushToRepositoryAction.kt

```diff
@@ -0,0 +1,365 @@
+        // valid tag is ascii letters, numbers, underscores, periods, or dashes
+        // https://docs.docker.com/engine/reference/commandline/tag/#extended-description
+        val validTagRegex = "[a-zA-Z0-9_.-]{1,128}".toRegex()
+
+        lateinit var fromLocalImageButton: JBRadioButton
+        lateinit var fromDockerfileButton: JBRadioButton
+
+        buttonGroup(::type) {
+            row {
+                fromLocalImageButton = this@row.radioButton(message("ecr.push.type.local_image.label"), BuildType.LocalImage).component
+                fromDockerfileButton = this@row.radioButton(message("ecr.push.type.dockerfile.label"), BuildType.Dockerfile).component
+            }
+        }
+
+        val imageSelectorPanel = localImageSelectorPanel()
+        val dockerfilePanel = dockerfileConfigurationSelectorPanel()
+
+        row {
+            cell(isFullWidth = true) {
+                imageSelectorPanel(grow)
+                    .installOnParent { fromLocalImageButton.isSelected }
+                    .visibleIf(fromLocalImageButton.selected)
+                dockerfilePanel(grow)
+                    .installOnParent { fromDockerfileButton.isSelected }
+                    .visibleIf(fromDockerfileButton.selected)
+            }
+        }
+
+        row(message("ecr.repo.label")) {
+            component(remoteRepos)
+                .growPolicy(GrowPolicy.MEDIUM_TEXT)
+                .withErrorOnApplyIf(message("ecr.repo.not_selected")) { it.selected() == null }
+        }
+
+        row(message("ecr.push.remoteTag")) {
+            textField(::remoteTag)
+                .constraints(grow)
+                .also {
+                    it.component.emptyText.text = defaultTag
+                }
+                .withErrorOnApplyIf(message("ecr.tag.invalid")) { it.text.isNotEmpty() && !it.text.matches(validTagRegex) }
+        }
+    }
+
+    private fun localImageSelectorPanel() = panel {
+        row(message("ecr.push.source")) {
+            // property binding syntax causes kotlin compiler error for some reason
+            comboBox(
+                localImageRepoTags,
+                { localImage },
+                { ::localImage.set(it) },
+                listCellRenderer { value, _, _ ->
+                    text = value.tag ?: value.imageId.take(15)
+                }
+            )
+                .growPolicy(GrowPolicy.MEDIUM_TEXT)
+                .withErrorOnApplyIf(message("ecr.image.not_selected")) { it.selected() == null }
+        }
+    }
+
+    private fun dockerfileConfigurationSelectorPanel() = panel {
+        row(message("ecr.dockerfile.configuration.label")) {
+            cell {
+                val model = CollectionComboBoxModel<DockerRunConfiguration>()
+                rebuildRunConfigurationComboBoxModel(model)
+                val box = comboBox(
+                    model,
+                    { runConfiguration },
+                    { ::runConfiguration.set(it) },
```

Comment:
Why is this not using `::runConfiguration` passed to `comboBox(
    model: ComboBoxModel<T>,
    prop: KMutableProperty0<T>,
    renderer: ListCellRenderer<T?>? = null
  ): CellBuilder<ComboBox<T>>`

Reply:
Oh, I see the comment earlier in file

Reply:
yeah it appears to only happen with the nullable fields, and the whole system around those is a bit wonky so it's actually cleaner like this